R/format.R

```diff
@@ -39,7 +39,7 @@ xportr_format <- function(.df, metacore, domain = NULL, verbose = getOption("xpo
   if (!is.null(attr(.df, "_xportr.df_arg_"))) df_arg <- attr(.df, "_xportr.df_arg_")
```

Comment:
Shall we increase the cyclomatic complexity to 20 or try and figure this one out?

Reply:
Changing lines 76-84 to the code below should solve the problem.

The second loop is not really needed.

```R
for (i in seq_len(ncol(.df))) {
    format_sas <- purrr::pluck(format, colnames(.df)[i])
    if (is.na(format_sas) || is.null(format_sas))
      format_sas <- ""
    attr(.df[[i]], "format.sas") <- format_sas
  }
```

from:

```R
  for (i in names(format)) {
    attr(.df[[i]], "format.sas")  <- format[[i]]
  }
  
  # Convert NA formats to "" for haven
  for (i in seq_len(ncol(.df))) {
    if (is.na(attr(.df[[i]], "format.sas")) || is.null(attr(.df[[i]], "format.sas"))) 
      attr(.df[[i]], "format.sas") <- ""
  }
```